src/cat.jl

```diff
@@ -4,7 +4,7 @@ function Base.cat(a::NamedDimsArray{L}; dims) where L
     data = Base.cat(parent(a); dims=numerical_dims) # Base.cat is type unstable
     T = eltype(a)  # therefore the element type has to be inferred manually
     N = length(newL)  # as must the size of the array
-    return NamedDimsArray{newL, T, N, Array{T,N}}(data)
+    return NamedDimsArray{newL, eltype(data), ndims(data), typeof(data)}(data)
 end
 
 # While the following two functions are covered by the general case below where splatted c
```

Comment:
Is this still true btw? This is what I get with or without these cases:
```
julia> nda = NamedDimsArray(rand(2,3), (:x, :y)); r23 = rand(2,3);

julia> @btime cat($nda, $r23, dims=:z);
  3.988 μs (33 allocations: 1.52 KiB)
```

Reply:
Right I was scratching my head about all the workarounds I did to make it not allocate that were just removed here without any effect. Potentially a result of the recent changes here https://github.com/invenia/NamedDims.jl/pull/132/files

Reply:
Ah could be, those look relevant. Shall I kill these?

Reply:
with fire

Reply:
Done!